cmake/leatherman.cmake

```diff
@@ -131,6 +131,88 @@ macro(enable_cpplint)
     endif()
 endmacro()
 
+# Usage: gettext_templates(dir)
+#
+# Create templates for gettext in `dir` from the source files found in additional specified dirs
+# Creates a custom target `translation`.
+macro(gettext_templates dir)
+    find_program(XGETTEXT_EXE xgettext)
+    find_program(MSGINIT_EXE msginit)
+    find_program(MSGMERGE_EXE msgmerge)
+
+    if (XGETTEXT_EXE AND MSGINIT_EXE AND MSGMERGE_EXE)
```

Comment:
Doing this with Make instead of CMake requires a cross-platform option for manipulating the file.